lib/default.nix

```diff
@@ -393,4 +393,96 @@ in {
   # one in the list will be used.
   combineFlakes = sep: prefixAndFlakes: builtins.foldl' addFlakes {}
     (lib.mapAttrsToList (prefix: flake: prefixFlake prefix sep flake) prefixAndFlakes);
+
+  mkFlake = project: packages: coverageProjectModule:
+    let
+      packageNames = project: builtins.attrNames (project.args.flake.packages project.hsPkgs);
+      checkedProject = project.appendModule { checkMaterialization = true; };
+      coverageProject = project.appendModule [
+        coverageProjectModule
+        {
+          modules = [{
+            packages = lib.genAttrs (packageNames project)
+              (_: { doCoverage = lib.mkDefault true; });
+          }];
+        }
+      ];
+      checks = builtins.listToAttrs (
+        lib.concatMap (packageName:
+          let package = project.hsPkgs.${packageName};
+          in lib.mapAttrsToList (n: v:
+              { name = "${packageName}:test:${n}"; value = v; })
+            (lib.filterAttrs (_: v: lib.isDerivation v) (package.checks))
+        ) (packageNames project));
+      ciJobs = {
+          # Run all the tests
+          inherit checks;
+          # Build tools and cache tools needed for the project
+          inherit (project) roots;
+          # Also build and cache any tools in the `devShell`
+          devShell = project.shell;
+          # Run HPC on the tests
+          coverage = builtins.listToAttrs (lib.concatMap (packageName: [{
```

Comment:
Do we not want to make the whole coverage thing optional? This seems to enable coverage by default on any project that uses this stuff? Maybe that's fine.

Reply:
I wondered about that too, but I was thinking there are already a couple of ways to pull out just what you want from the result.  `removeAttrs` or `filterAttrs` and you can also just have `{ inherits (f) packages checks; }`.  It does get a little trickier with the nested attributes like `coverage` though.  Happy to add something if you think it is worth it.

I'm also wondering how we select `required` hydra jobs.   Maybe we need a way to set `coverageDisabled` and `coverageRequired` on the test components?

Reply:
I think wanting coverage is usually a per-project toggle that you set when you're thinking about CI, so having it as an argument to the `ciJobs` function would be reasonable, `enableCoverage :: Bool`, default false. Yes, it means we have to pipe it up to the flake arguments or whatever...

This is even simpler if we make things more top-level functions and let people put them together themselves.

```
projectCiJobs = project: ...
projectCoverageCiJobs = project: ...
myCiJobs = projectCiJobs myProject // projectCoverageCiJobs myProject
myFlake = {
  hydraJobs = myCiJobs
}
```

Reply:
Does `mkFlakeCiJobs` look ok now?  My thinking was that you can pass in `coverage = {}` or `coverage = projectCoverageCiJobs  project selectPackages coverageProjectModule`.